files/galaxy-test/tpv/tool_defaults.yml

```diff
@@ -25,6 +25,17 @@ tools:
           training_labels = '"' + ", ".join(training_roles) + '"'
           entity.params['requirements'] = '(GalaxyGroup == "compute") || (%s)' % training_expr if training_expr else '(GalaxyGroup == "compute")'
           entity.params['+Group'] = training_labels
+      - id: remote_resources
+        if: user is not None
+        execute: |
+          user_preferences = user.extra_preferences
+          for data_item in user_preferences:
+            if "distributed_compute|remote_resources" in data_item:
```

Comment:
should it be different because of TPV? I would think it is on Galaxy side

Reply:
It is I'm just wondering... You are checking here for a key in a dict?

Reply:
yes for these keys, as far as I understand: `distributed_compute|remote_resources`
https://github.com/usegalaxy-eu/infrastructure-playbook/blob/a799b2d3cc026eb2587909f0ef224e9e0c9f7725/files/galaxy/config/user_preferences_extra_conf.yml#L35-L39

I copied the code over from sorting-hat. I think user.extra_preferences is a list.
Here is similar code in the galaxy backend: https://github.com/galaxyproject/galaxy/blob/b58e165002ddbfdf2ca74e1aa048dabfba93c4b1/lib/galaxy/webapps/galaxy/api/users.py#L509
and here in sorting-hat: https://github.com/usegalaxy-eu/sorting-hat/blob/0b0758a1b8b72bc0ea5ae198ad2949f4aa16b586/sorting_hat.py#L367-L371

Reply:
https://github.com/usegalaxy-eu/sorting-hat/blob/0b0758a1b8b72bc0ea5ae198ad2949f4aa16b586/sorting_hat.py#L367-L371
https://github.com/galaxyproject/galaxy/blob/b58e165002ddbfdf2ca74e1aa048dabfba93c4b1/lib/galaxy/webapps/galaxy/api/users.py#L509-L511
I don't know if this is correct

Reply:
> should it be different because of TPV? I would think it is on Galaxy side

I also think it should be the same as it was for sorting hat, as it is coming from a User object.